test/parsers/parameters.test.ts

```diff
@@ -0,0 +1,172 @@
+import { assertEquals, assertThrows } from '@std/assert';
+
+import { ParseParenthesized } from '../../src/parsers/parameters.ts';
+
+Deno.test('ParseParenthesizedList - Basic functionality', () => {
+  // Simple case
+  assertEquals(ParseParenthesized('(a b c)')?.val, ['a', 'b', 'c']);
+
+  // Empty parentheses
+  assertEquals(ParseParenthesized('()')?.val, []);
+
+  // Single item
+  assertEquals(ParseParenthesized('(hello)')?.val, ['hello']);
+
+  // No opening parenthesis
+  assertEquals(ParseParenthesized('hello world')?.val, 'hello');
+});
+
+Deno.test('ParseParenthesizedList - Nested structures', () => {
+  // Simple nesting
+  assertEquals(
+    ParseParenthesized('(a (b c) d)')?.val,
+    ['a', ['b', 'c'], 'd'],
+  );
+
+  // Deep nesting
+  assertEquals(
+    ParseParenthesized('(a (b (c d) e) f)')?.val,
+    ['a', ['b', ['c', 'd'], 'e'], 'f'],
+  );
+
+  // Multiple nested groups
+  assertEquals(
+    ParseParenthesized('((a b) (c d))')?.val,
+    [['a', 'b'], ['c', 'd']],
+  );
+});
+
+Deno.test('ParseParenthesizedList - Quoted strings', () => {
+  // Basic quoted string
+  assertEquals(
+    ParseParenthesized('("hello world" test)')?.val,
+    ['"hello world"', 'test'],
+  );
+
+  // Quoted string with escaped quotes
+  assertEquals(
+    ParseParenthesized('("hello \\"world\\"" test)')?.val,
+    ['"hello \\"world\\""', 'test'],
+  );
+
+  // Mixed quoted and unquoted
+  assertEquals(
+    ParseParenthesized('(unquoted "quoted string" more)')?.val,
+    ['unquoted', '"quoted string"', 'more'],
+  );
+});
+
+Deno.test('ParseParenthesizedList - IMAP literals', () => {
+  // Basic literal
+  assertEquals(
+    ParseParenthesized('({11}\r\nhello world)')?.val,
+    ['"hello world"'],
+  );
+
+  // Literal with LF only
+  assertEquals(
+    ParseParenthesized('({5}\nhello world)')?.val,
+    ['"hello"', 'world'],
+  );
+
+  // Multiple literals
+  assertEquals(
+    ParseParenthesized('({3}\r\nabc {2}\r\nxy)')?.val,
+    ['"abc"', '"xy"'],
+  );
+
+  // Literal with special characters
+  assertEquals(
+    ParseParenthesized('({10}\r\nhello (world)')?.val,
+    ['"hello (wor"', 'ld'],
+  );
+});
+
+Deno.test('ParseParenthesizedList - IMAP ENVELOPE example', () => {
+  const envelope = '(NIL "Subject Line" ((NIL NIL "user" "example.com") NIL) NIL NIL NIL)';
```

Comment:
Ah, this helps, so in the case where a value is omitted, i'm guessing the spec must dictate NIL. TIL